Keithley2612B_voltage_sweep.py

```diff
@@ -164,36 +235,41 @@ def sweep_operation(smu_id, steps_no, measure_delay, nplc, start, end):
 
             target_volt = get_target_volt(start_volt)
             
-            steps_num = int (input ('Enter the number of steps'))
+            steps_num = int (input ('Enter the number of steps: '))
 
             integration_time = get_integration_time()
             
-            measure_delay = float (input('Set settling delay (in seconds) before each measurement: (NOTE: Setting the value to -1 automatically starts taking measurement as soon as current is stable) '))
+            measure_delay = 0
 
             #pulsed = get_sweep_type()
 
+            scan_rate = float(input ('Enter scan rate in mV/s: '))
             
             # VI measurement
             print ('Voltage sweep being conducted...')
-            test_output = sweep_operation (smu, steps_num, measure_delay, integration_time, start_volt, target_volt) 
+            test_output = sweep_operation (smu, steps_num, measure_delay, integration_time, start_volt, target_volt, scan_rate) 
             end_time = datetime.now()
             print (f"Sweep successfully completed on {end_time}.")
 
             # scan rate 
             # (Unsure of the data type of measured voltage and timestamps. This will only work if both are float or int)   
     
-            del_t = []
+            del_t = np.empty((1,steps_num))
+            print("The test_output value is: " + str(test_output[0]))
+            print("The length is: " + str(len(test_output[2])))
             for i in range(len(test_output[2])):
-                del_t [i] = test_output [2][i] - test_output [2][0]
+                del_t[0][i] = test_output[2][i] - test_output[2][0]
```

Comment:
Why is del_t a 2-D array?

Reply:
I am not sure. del_t was created as a numpy array, and somehow despite being 1D, it kept throwing an error when I used only one [i]. I threw in the [0] and suddenly it worked. Python seems to think that del_t is 2D. 

I think it might be related to the declaration of del_t, where I made a numpy empty array with dimensions 1x3. I think because of that, del_t was recognised as a 2D array rather than 1D "list". But I am not sure. Maybe if all I did was to create a Python list, then this ambiguity would be resolved.